src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -351,6 +357,20 @@ internal static async Task ProcessSarifLogAsync(Stream stream, bool showMessageO
             if (sarifLog != null)
             {
                 await ProcessSarifLogAsync(sarifLog, logFilePath: null, showMessageOnNoResults: showMessageOnNoResults, cleanErrors: cleanErrors, openInEditor: openInEditor);
+                if (sarifLog.HasNoResults())
+                {
+                    new InfoBar(Resources.InfoNoResultsInLog, imageMoniker: KnownMonikers.StatusInformation).ShowAsync().FileAndForget(FileAndForgetEventName.InfoBarOpenFailure);
+                }
+
+                if (sarifLog.HasErrorLevelToolConfigurationNotifications())
+                {
+                    new InfoBar(Resources.ErrorLogHasErrorLevelToolConfigurationNotifications).ShowAsync().FileAndForget(FileAndForgetEventName.InfoBarOpenFailure);
+                }
+
+                if (sarifLog.HasErrorLevelToolExecutionNotifications())
+                {
+                    new InfoBar(Resources.ErrorLogHasErrorLevelToolExecutionNotifications).ShowAsync().FileAndForget(FileAndForgetEventName.InfoBarOpenFailure);
```

Comment:
Is it possible to have a sarif log with no results and the tool configuration contains an error?
Why: if that happen, we will show many issues in the `InfoBar`. Is that expected?

Reply:
Yes, it is possible to have such a log. In fact it is likely: If the tool detects a catastrophic configuration error, it probably won't produce results. And you're right, we don't want multiple bars in that case.

But if we are sent two logs, and one has "no results" and the other has "catastrophic configuration error", then IMO we _should_ show one of each kind of bar. What we don't want is 50 "no results" bars or 50 "catastrophic configuration error" bars. That's why the messages talk about "one or more SARIF logs".

See also my answer to Gabe. Your two comments, taken together, suggest some additional logic, and a refactoring to separate out that logic and make it unit testable, which I will now do.

---
In reply to: [523400552](https://github.com/microsoft/sarif-visualstudio-extension/pull/272#discussion_r523400552) [](ancestors = 523400552)

Reply:
This is now done.

---
In reply to: [523445875](https://github.com/microsoft/sarif-visualstudio-extension/pull/272#discussion_r523445875) [](ancestors = 523445875,523400552)